core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -160,6 +164,21 @@ private static Optional<String> getBusinessMetricsString(ExecutionAttributes exe
         return Optional.of(businessMetrics.asBoundedString());
     }
 
+    private static void checksumBusinessMetrics(ExecutionAttributes executionAttributes,
+                                                   BusinessMetricCollection businessMetrics) {
+        BusinessMetricsUtils.resolveRequestChecksumCalculationMetric(
+            executionAttributes.getAttribute(SdkInternalExecutionAttribute.REQUEST_CHECKSUM_CALCULATION))
+            .ifPresent(businessMetrics::addMetric);
+        
+        BusinessMetricsUtils.resolveResponseChecksumValidationMetric(
+            executionAttributes.getAttribute(SdkInternalExecutionAttribute.RESPONSE_CHECKSUM_VALIDATION))
+            .ifPresent(businessMetrics::addMetric);
+        
+        BusinessMetricsUtils.resolveChecksumSpecsMetric(
+            executionAttributes.getAttribute(SdkExecutionAttribute.RESOLVED_CHECKSUM_SPECS))
```

Comment:
Hmm, RESOLVED_CHECKSUM_SPECS is resolved in `HttpChecksumStage`, which happens after `ApplyUserAgentStage`

Reply:
Yeah, it seems brittle. Let's get ApplyUserAgentStage moved first